src/main/scala/advent/solutions/Day3.scala

```diff
@@ -81,17 +240,120 @@ object Day3 {
       *         or [[None]] if they don't intersect at all
       */
     def numberOfStepsToIntersection(wire0: Wire, wire1: Wire): Option[Int] = {
-      ???
+
+      val out = intersectionPoints(wire0, wire1) match {
+        case Some(x) => {
+          val result = x.map(intersectionPoint => {
+            countNumberOfStepsToAnIntersection(wire0, intersectionPoint) + countNumberOfStepsToAnIntersection(
+              wire1,
+              intersectionPoint
+            )
+          })
+          Some(result.min)
+        }
+        case None => None
+      }
+      out
+    }
+
+    /**
+      *
+      * @param wire
+      * @param intersectionPoint
+      * @return
+      */
+    def countNumberOfStepsToAnIntersection(
+        wire: Wire,
+        intersectionPoint: Coordinate
+    ): Int = {
+      val turningPoints: List[Coordinate] = findTurningPoints(wire)
+
+      val startingAndEndingPoints
+          : List[(Displacement, StartingAndEndingPoints)] = wire.path
+        .zip(turningPoints.init)
+        .zip(turningPoints.tail)
+        .map(d => (d._1._1, StartingAndEndingPoints(d._1._2, d._2)))
+
+      val allCoordinates: List[(Displacement, List[Coordinate])] =
+        startingAndEndingPoints.map(i => (i._1, pathTravelled(i._1, i._2)))
+
+      val selectedDisplacement = allCoordinates
```

Comment:
I think the list of coordinates up until the intersection point should have the same size as the number of steps taken to reach it.  It might be simpler to do a `takeWhile` until you find the `intersectionPoint` and then find the size of the list.
Then you wouldn't need to bother examining at the `startingAndEndingPoints` or `allPreviousSteps`.